src/resources/js/attendee-data-editor.js

```diff
@@ -0,0 +1,169 @@
+( function( $ ) {
+	var editing = false;
+	var event_id = 0;
+	var product_id = 0;
+	var attendee_id = 0;
+	var $row;
+	var $editor;
+	var $editor_inner;
+	var $editor_close;
+	var $save_btn;
+
+	function init() {
+		create_inline_editor_space();
+		$( 'table.attendees' ).on( 'click', '.edit-attendee-data', on_edit );
+	}
+
+	function on_edit() {
+		// Already editing? Do nothing
+		if ( editing ) {
+			return false;
+		}
+
+		editing = true;
+		open_for_editing( $( this ) );
+		return false;
+	}
+
+	function open_for_editing( $trigger ) {
+		$row = $trigger.parents( 'tr' );
+		event_id = $trigger.data( 'event-id' );
+		product_id = $trigger.data( 'product-id' );
+		attendee_id = $trigger.data( 'attendee-id' );
+
+		$editor.fadeIn( 'fast' );
+		request_attendee_form();
+	}
+
+	function create_inline_editor_space() {
+		$editor = $(
+			  '<div id="attendee-data-editor-backscreen">'
+			+ '<div id="attendee-data-editor-space">'
+			+ '<div class="close"> &#10060; </div>'
+			+ '<div class="inner"></div>'
+			+ '</div> </div>'
+		);
+
+		$editor_inner = $editor.find( 'div.inner' );
+		$editor_close = $editor.find( 'div.close' );
+		$editor_close.on( 'click', terminate_editor );
+
+		$editor.hide();
+		$( 'body' ).append( $editor );
+	}
+
+	function terminate_editor() {
+		$editor.fadeOut( 'fast', function() {
+			$editor_inner.html( '' );
+			editing = false;
+		} );
+	}
+
+	function request_attendee_form() {
+		$.post( ajaxurl, {
+				'action':      'load_attendee_data_editor_form',
+				'check':       attendee_data_editor.check,
+				'event_id':    event_id,
+				'attendee_id': attendee_id
+			},
+			load_attendee_form,
+			'json'
+		);
+	}
+
+	function load_attendee_form( data ) {
+		if (
+			'object' !== typeof data
+			|| 'undefined' === typeof data.success
+			|| ! data.success
+			|| 'object' !== typeof data.data
+			|| 'string' !== typeof data.data.html
+		) {
+			console.error( 'Attendee Data Editor: something went wrong when we tried to fetch the data editor content' );
```

Comment:
could be made translatable but don't have to for this version

Reply:
Stripped this out.